### ISSUE TITLE
[Bug]: Unable to create a foreign key constraint via GUI due to missing target table/column option.

### ISSUE THREAD

**Issue by pchemguy** [sqlitebrowser/sqlitebrowser#3218]
### What did you do?

  1. Create a new in-memory database.
  2. Populate it using this SQL:

~~~sql
DROP TABLE IF EXISTS paths;
CREATE TABLE paths (
    prefix  TEXT,
    name    TEXT,
    paths   TEXT GENERATED ALWAYS AS (prefix || name)
);
INSERT INTO paths(prefix, name) VALUES ('/user/bin/', 'bash');

DROP TABLE IF EXISTS prefixes;
CREATE TABLE prefixes (
    prefix  TEXT PRIMARY KEY
);
INSERT INTO prefixes(prefix) VALUES ('/user/bin/');
~~~

  4. Attempt to create a foreign key paths.prefix -> prefixes.prefix (Modify Table -> Constraints -> Add constraint -> Foreign Key.

### What did you expect to see?

A form to specify the source and target for the new foreign key.

### What did you see instead?

The resultant form does not provide the ability to specify the target (table/column). This is an old issue; I do not recall ever being able to do this (I used SQLiteStudio for this purpose).

![fk](https://user-images.githubusercontent.com/39730837/200162682-0c0d109a-db1e-444a-9570-8ef0cd5e3a0e.jpg)

### DB4S Version

3.12.2

### What OS are you seeing the problem on?

Windows

### OS version

Windows 10

### Prevention against duplicate issues

- [X] I have searched for similar issues

**Comment by chrisjlocke** [comment]
Its there, it's just a little .... quirky.

Modify the table, 'Paths'.  

![image](https://user-images.githubusercontent.com/7872218/200163564-f1998654-cb45-41cd-8b9a-aa6e9e2d5be1.png)

Scroll across (1) and you'll see 'Foreign Keys' (2)


![image](https://user-images.githubusercontent.com/7872218/200163721-1477104a-d3c2-4ddf-8db3-e03076ec6e26.png)

Click in 'Foreign Key' (1) and you get a table dropdown.  Select the other table (prefixes)


![image](https://user-images.githubusercontent.com/7872218/200163626-f3fa462e-09c4-4f6c-a49c-224bcc276d1c.png)

This populates the second dropdown of fields (1), so select 'prefix'


![image](https://user-images.githubusercontent.com/7872218/200163771-93fd8552-0ca7-48dc-9de2-21cb53291dcc.png)

This is the sort of quirky bit.  You have to click elsewhere, eg, where the (1) is, for the entry to 'save'.  If you click OK, the entry hasn't been populated fully, and it may not register.  You'll see the entry plop in, so then click OK.

**However**

![image](https://user-images.githubusercontent.com/7872218/200163844-3b43d186-ce51-4c0b-a9b7-72a8d6a11788.png)

This will of course fail due to the generated column.

**Comment by chrisjlocke** [comment]
Had this of worked, you'll then see it on the constaints tab, where you can set a name.

![image](https://user-images.githubusercontent.com/7872218/200163965-f7f4d15d-f144-4cd8-8a87-2f3666f3fb32.png)

**Comment by pchemguy** [comment]
Right, I thought that once I was able to somehow create it in DB4S. The the question is, if FK cannot be created via the "Add constraint", then why is it there in the first place?

**Comment by chrisjlocke** [comment]
I think its to ... er, I believe we ... it may be because ...

No, I have no idea. :(

**Comment by synapticweb** [comment]
If it weren't for this issue, I would never have found out how to create a foreign key in sqlitebrowser. I lost like half an hour staring at the screen. It's unbelievable.
